commit cfcd3d3974ab0846e8e0ffb8a41c19e5c170620d
Author: Jeff Law <jlaw@ventanamicro.com>
Date:   Wed Apr 2 14:30:34 2025 -0600

    Revert "RISC-V: Make VXRM as global register [PR118103]"
    
    This reverts commit 1c8e6734d2dd3a6236d94c6e4e0c6780f35ede9f.

diff --git a/gcc/config/riscv/riscv.cc b/gcc/config/riscv/riscv.cc
index 38f3ae7cd84..36a210f02fc 100644
--- a/gcc/config/riscv/riscv.cc
+++ b/gcc/config/riscv/riscv.cc
@@ -10914,9 +10914,7 @@ riscv_conditional_register_usage (void)
 	call_used_regs[regno] = 1;
     }
 
-  if (TARGET_VECTOR)
-    global_regs[VXRM_REGNUM] = 1;
-  else
+  if (!TARGET_VECTOR)
     {
       for (int regno = V_REG_FIRST; regno <= V_REG_LAST; regno++)
 	fixed_regs[regno] = call_used_regs[regno] = 1;
